src/Components/Components/src/BindConverter.cs

```diff
@@ -0,0 +1,1047 @@
+                    }
+                    else if (typeof(T) == typeof(DateTime))
+                    {
+                        parser = ConvertToDateTime;
+                    }
+                    else if (typeof(T) == typeof(DateTime?))
+                    {
+                        parser = ConvertToNullableDateTime;
+                    }
+                    else if (typeof(T) == typeof(DateTimeOffset))
+                    {
+                        parser = ConvertToDateTime;
+                    }
+                    else if (typeof(T) == typeof(DateTimeOffset?))
+                    {
+                        parser = ConvertToNullableDateTime;
+                    }
+                    else if (typeof(T).IsEnum)
+                    {
+                        // We have to deal invoke this dynamically to work around the type constraint on Enum.TryParse.
+                        var method = _convertToEnum ??= typeof(BindConverter).GetMethod(nameof(ConvertToEnum), BindingFlags.NonPublic | BindingFlags.Static);
+                        parser = method.MakeGenericMethod(typeof(T)).CreateDelegate(typeof(BindParser<T>), target: null);
+                    }
+                    else if (Nullable.GetUnderlyingType(typeof(T)) is Type innerType && innerType.IsEnum)
+                    {
+                        // We have to deal invoke this dynamically to work around the type constraint on Enum.TryParse.
+                        var method = _convertToNullableEnum ??= typeof(BindConverter).GetMethod(nameof(ConvertToNullableEnum), BindingFlags.NonPublic | BindingFlags.Static);
+                        parser = method.MakeGenericMethod(innerType).CreateDelegate(typeof(BindParser<T>), target: null);
+                    }
+                    else
+                    {
+                        parser = MakeTypeConverterConverter<T>();
+                    }
+
+                    _cache.TryAdd(typeof(T), parser);
+                }
+
+                return (BindParser<T>)parser;
+            }
+
+            private static BindParser<T> MakeTypeConverterConverter<T>()
+            {
+                var typeConverter = TypeDescriptor.GetConverter(typeof(T));
+                if (typeConverter == null || !typeConverter.CanConvertFrom(typeof(string)))
+                {
+                    throw new InvalidOperationException(
+                        $"The type '{typeof(T).FullName}' does not have an associated {typeof(TypeConverter).Name} that supports " +
+                        $"conversion from a string. " +
+                        $"Apply '{typeof(TypeConverterAttribute).Name}' to the type to register a converter.");
+                }
+
+                return ConvertWithTypeConverter;
+
+                bool ConvertWithTypeConverter(object obj, CultureInfo culture, out T value)
+                {
+                    var text = (string)obj;
+                    if (string.IsNullOrEmpty(text))
```

Comment:
Just because I don't know TypeConverter conventions as well as you do, is it *always* the case that empty string maps to `default`? Might someone ever want to write a TypeConverter that maps empty string to something else, or is that problematic for some reason?

Reply:
Yeah, it's a good point this should probably allow the type converter to interpret the value.